tests/test_mongodb_id_pk_variations.py

```diff
@@ -0,0 +1,228 @@
+import tap_tester.connections as connections
+import tap_tester.menagerie   as menagerie
+import tap_tester.runner      as runner
+import os
+import unittest
+import pymongo
+import string
+import random
+import time
+import datetime
+from mongodb_common import drop_all_collections, get_test_connection, ensure_environment_variables_set
+import copy
+
+RECORD_COUNT = {}
+
+
+def random_string_generator(size=6, chars=string.ascii_uppercase + string.digits):
+    return ''.join(random.choice(chars) for x in range(size))
+
+
+def generate_docs_no_id(num_docs):
+    docs = []
+    for int_value in range(num_docs):
+        docs.append({"int_field": int_value, "string_field": random_string_generator()})
+    return docs
+
+
+def generate_docs_int_id(num_docs):
+    docs = []
+    for int_value in range(num_docs):
+        docs.append({"_id": int_value, "string_field": random_string_generator()})
+    return docs
+
+
+def generate_docs_double_id():
+    docs = []
+    docs.append({"_id": 546.43, "string_field": random_string_generator()})
+    docs.append({"_id": 546.56, "string_field": random_string_generator()})
+    return docs
+
+
+def generate_docs_string_id():
+    docs = []
+    docs.append({"_id": 'primary_key', "string_field": random_string_generator()})
+    docs.append({"_id": 'secondary_key', "string_field": random_string_generator()})
+    return docs
+
+
+def generate_docs_binary_id():
+    docs = []
+    docs.append({"_id": '10101011', "string_field": random_string_generator()})
```

Comment:
I think this may just be a string of 1's and 0's and not actually a binary data type. For python you can represent a binary value as an integer  of 1's and 0's prefixed by `0b` so  instead of `'10101011'` it would be `0b10101011`. There are other ways to represent this like using `bytes` and `bytearray` functions. I would check out the pymongo documentations to verify each of these datatypes will be encoded as you expect them to. Here is one page I found for using bson (binary json) https://pymongo.readthedocs.io/en/stable/api/bson/index.html.

Reply:
done